Meta/Lagom/Tools/CodeGenerators/JSSpecCompiler/Parser/CppASTConverter.cpp

```diff
@@ -175,4 +177,32 @@ Tree CppASTConverter::as_possibly_empty_tree(Cpp::Statement const* statement)
     return make_ref_counted<TreeList>(Vector<Tree> {});
 }
 
+CppParsingStep::CppParsingStep()
+    : CompilationStep("parser"sv)
+{
+}
+
+CppParsingStep::~CppParsingStep() = default;
+
+void CppParsingStep::run(TranslationUnitRef translation_unit)
+{
+    auto filename = translation_unit->filename;
+
+    auto file = Core::File::open_file_or_standard_stream(filename, Core::File::OpenMode::Read).release_value_but_fixme_should_propagate_errors();
+    m_input = file->read_until_eof().release_value_but_fixme_should_propagate_errors();
```

Comment:
after the run function, this m_input variable will be a dangling reference, as you don't hold onto the file object.

Should it really be a member?

Reply:
`m_input` is a `ByteBuffer`, which is an owner of all string views we have in ASTs (and this is actually the primary reason why I keep all of the compilation step classes until compilation finishes)